Client/Frontend/Browser/URLBarView.swift

```diff
@@ -182,7 +182,11 @@ class URLBarView: UIView, BrowserLocationViewDelegate, UITextFieldDelegate {
     }
 
     func textFieldDidBeginEditing(textField: UITextField) {
-        textField.selectAll(nil)
+        // Without the async dispatch below, the selectAll() doesn't work
+        // intermittantly, and also crashes on iPhone 6 Plus (bug 1124310).
+        dispatch_async(dispatch_get_main_queue(), {
+            textField.selectAll(nil)
```

Comment:
Don't use `selectAll()` .. use this instead:
```
textField.selectedTextRange = textField.textRangeFromPosition(textField.beginningOfDocument,
    toPosition: textField.endOfDocument)
```

According to the documentation `selectAll()` is part of `UIResponderStandardEditActions` but is not supported in Swift. It also is a much higher level thing it seems, that `UIResponder` subclasses should implement to support the Select All option from the text context menu.